pinot-connectors/pinot-flink-connector/src/test/java/org/apache/pinot/connector/flink/http/PinotControllerClientTest.java

```diff
@@ -0,0 +1,168 @@
+
+    when(client.target(expectedFullURL)).thenReturn(target);
+    when(target.request()).thenReturn(builder);
+    when(builder.headers(headers)).thenReturn(builder);
+    when(builder.get()).thenReturn(response);
+    when(response.getStatus()).thenReturn(200);
+    when(response.getEntity()).thenReturn(resEntity);
+    when(response.readEntity(Map.class)).thenReturn(resEntity);
+
+    PinotControllerClient controllerClient = new PinotControllerClient();
+    controllerClient.setHttpClient(client);
+    List<String> instances = controllerClient.getControllerInstances(new MultivaluedHashMap<>());
+    assertEquals(Lists.newArrayList("pinot-prod02:5983"), instances);
+
+    verify(client, times(1)).target(expectedFullURL);
+    verify(target, times(1)).request();
+    verify(builder, times(1)).headers(headers);
+    verify(builder, times(1)).get();
+  }
+
+  @Test
+  public void testGetPinotSchemaStrFromController()
+      throws Exception {
+    Client client = mock(Client.class);
+    WebTarget target = mock(WebTarget.class);
+    Invocation.Builder builder = mock(Invocation.Builder.class);
+    Response response = mock(Response.class);
+    MultivaluedMap<String, Object> headers = getRequestHeadersToPinot();
+    String expectedFullURL = "http://localhost:9000/schemas/demand";
+    final Map<String, Object> resEntity =
+        JsonUtils.stringToObject(fixture("fixtures/pinotTableSchema.json"), Map.class);
+
+    when(client.target(expectedFullURL)).thenReturn(target);
+    when(target.request()).thenReturn(builder);
+    when(builder.headers(headers)).thenReturn(builder);
+    when(builder.get()).thenReturn(response);
+    when(response.getStatus()).thenReturn(200);
+    when(response.getEntity()).thenReturn(resEntity);
+    when(response.readEntity(Map.class)).thenReturn(resEntity);
+
+    PinotControllerClient controllerClient = new PinotControllerClient();
+    controllerClient.setHttpClient(client);
+    String schemaStrFromController =
+        controllerClient.getSchemaStrFromController(TABLE_NAME, new MultivaluedHashMap<>());
+    assertEquals(
+        JsonUtils.objectToString(JsonUtils.stringToObject(fixture("fixtures/pinotTableSchema.json"), Map.class)),
+        schemaStrFromController);
+
+    verify(client, times(1)).target(expectedFullURL);
+    verify(target, times(1)).request();
+    verify(builder, times(1)).headers(headers);
+    verify(builder, times(1)).get();
+  }
+
+  @Test
+  public void testGetPinotConfigStrFromController()
+      throws Exception {
+    Client client = mock(Client.class);
+    WebTarget target = mock(WebTarget.class);
+    Invocation.Builder builder = mock(Invocation.Builder.class);
+    Response response = mock(Response.class);
+    MultivaluedMap<String, Object> headers = getRequestHeadersToPinot();
+    String expectedFullURL = "http://localhost:9000/tables/demand?type=realtime";
+    final Map<String, Object> resEntity = new HashMap<>() {
```

Comment:
This seems not compilable for jdk8:
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project flink-pinot-sink: Compilation failure
Error:  /home/runner/work/pinot/pinot/pinot-connectors/pinot-flink-connector/src/test/java/org/apache/pinot/connector/flink/http/PinotControllerClientTest.java:[124,53] error: cannot infer type arguments for HashMap<K,V>
Error:    reason: cannot use '<>' with anonymous inner classes
Error:    where K,V are type-variables:
Error:      K extends Object declared in class HashMap
Error:      V extends Object declared in class HashMap
Error:  -> [Help 1]
```

Reply:
I thought we dropped jdk8 support? not yet?

Reply:
btw, this usage `HashMap<>` is all over the codebase